src/parser.ts

```diff
@@ -167,17 +175,20 @@ function handleToken(
   const isLeafNode = typeName === 'image';
   if (!isLeafNode) {
     for (const child of token.children)
-      handleToken(child, nodes, file, inlineParent);
+      handleToken(child, nodes, file, handleSlots, inlineParent);
   }
 
   nodes.pop();
 }
 
-export default function parser(tokens: Token[], file?: string) {
+export default function parser(tokens: Token[], args?: string | ParserArgs) {
   const doc = new Node('document');
   const nodes = [doc];
 
-  for (const token of tokens) handleToken(token, nodes, file);
+  if (typeof args === 'string') args = { file: args };
+
+  for (const token of tokens)
+    handleToken(token, nodes, args?.file, args?.slots);
```

Comment:
Should `handleToken` just take `args` as a param?

Reply:
I think the implementation is a little cleaner this way. It's an internal function, so I don't really need it to match the user-facing API.